cool-demos/conditional-blocks.html

```diff
@@ -0,0 +1,452 @@
+
+                // The init event is fired when the editor is fully loaded.
+                // https://www.tiny.cloud/docs/advanced/events/#editorcoreevents
+                editor.on("init", () => {
+                    // Get the iframe window object and the iframes document object
+                    // and call our setup function that creates the web component
+                    // https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/#getwin
+                    // https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/#getdoc
+                    const win = editor.getWin();
+                    const doc = editor.getDoc();
+                    setup(win, doc, editor);
+                });
+
+                // The preinit event is fired after the editor is loaded but before
+                // the content is loaded
+                // https://www.tiny.cloud/docs/advanced/events/#editorcoreevents
+                editor.on("preinit", () => {
+                    // During the creation of the web component we set contenteditable false
+                    // on the web component to make it behave like a noneditable but selectable
+                    // element inside TinyMCE. But we don't want the contenteditable attribute
+                    // to be saved with the content. We therefore need to filter out the attribute
+                    // upon serlialization (which happens on "save", view sourcecode and preview
+                    // among others).
+                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.serializer/#addnodefilter
+                    editor.serializer.addNodeFilter("conditional-block", (nodes) => {
+                        // Iterate through all filtered nodes and remove the contenteditable attribute
+                        // as well as the wrapper <div> we add when creating the custom element
+                        nodes.forEach((node) => {
+                            if (!!node.attr("contenteditable")) {
+                                // Remove the contenteditable attribute on the <conditional-block>
+                                node.attr("contenteditable", null);
+                                // Traverse into the first child which would be the <div contenteditable="true">
+                                // https://www.tiny.cloud/docs/api/tinymce.html/tinymce.html.node/#walk
+                                const content = node.walk();
```

Comment:
This might be better using `node.firstChild` maybe? As this really doesn't need to use walk to just get the first child. It could also cause it to get the next sibling instead if the DOM is manipulated elsewhere.